src/toil/provisioners/aws/awsProvisioner.py

```diff
@@ -199,14 +223,20 @@ def __init__(self, clusterName, clusterType, zone, nodeStorage, nodeStorageOverr
         # or readClusterSettings()
         super().__init__(clusterName, clusterType, zone, nodeStorage, nodeStorageOverrides)
 
+        if self._zone is None:
+            logger.warning("Leader zone was never initialized before creating AWS provisioner. Defaulting to cluster zone.")
```

Comment:
Maybe this should be about how a zone was not provided (by the user) instead of never initialized (by us)?

Reply:
I think the RuntimeError above probably catches whether the zone is None, and the leader_subnet seems to just be a mirror of zone, which would be caught if it was None. So this logger message probably would never be reached. I can probably get rid of this logger message which I believe I contained originally for mypy.